Fixes ENYO-2144

ryanjduffy opened the pull request:
## Issue

Expandable controls are duplicating component tree and features
## Fix

Refactor things for better reuse

Enyo-1.1-DCO-Signed-off-by: Ryan Duffy ryan.duffy@lge.com

Commit:
refactor expandable pickers for better reuse

Issue: ENYO-2144
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

ExpandableListItem.js: Wrapped CSS in a .moon-expandable-list-item class; removed 3 superfluous rules due to moon-item already setting them; fixed RTL status arrow position; simplified spotlight color and arrow logic;

ExpandableListItem.js: Changed currentValueShowing to only display the currentValue if it has content. This fixes a positioning issue caused by an empty currentValue control. (This approach may need revision)

Enyo-DCO-1.1-Signed-off-by: Blake Stephens <blake.stephens@lge.com>

reconnect ExpandableInput's ondown handler

Issue: ENYO-2144
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

refactor LabeledListItem to be the expandables header

Issue: ENYO-2144, ENYO-1798
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

add ExpandableListItemHeader

To avoid adding unnecessary API to LabeledTextItem, I've added a new
private kind to ExpandableListItem that adds the customization to
LabeledTextItem. This also simplified the API a bit as one of the
properties (textWrapped) could be removed.

Issue: ENYO-2144
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

add CSS rule for LabeledTextItem within .moon-neutral container

Issue: ENYO-2144, ENYO-2155
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

few more fixes

Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

rework 5-way spotlight handlers in ExpandableListItem

Issue: ENYO-2284
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

deprecate moonstone/Accordion

With the refactoring of expandables, the Accordion control is identical
to ExpandableListItem with the only addition being a CSS class without
any rules.

Issue: ENYO-2144
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

aarontam commented on lib/ExpandableInput/ExpandableInput.js:
```diff
@@ -180,11 +163,12 @@ module.exports = kind(
 	},
 
 	/**
+<<<<<<< HEAD
```
Some merge conflict remnants. :smile:

aarontam commented on lib/Accordion/Accordion.js:
```diff
@@ -89,33 +81,5 @@ module.exports = kind(
-	bindings: [
-		{from: '.disabled', to: '.$.headerWrapper.disabled'}
-	]
+	classes: 'moon-accordion'
```
I wonder if, somewhere down the road, we can deprecate this kind and just make this a special case of `ExpandableListItem` i.e. enabling via a property of that kind.

ryanjduffy replied:
Agreed. Kept it but marked it deprecated.

aarontam commented on lib/ExpandableInput/ExpandableInput.js:
```diff
@@ -294,29 +266,11 @@ module.exports = kind(
-	},
-
-	/**
-	* We manually set pointer mode to `false` as this seems to be the
```
Glad we could get rid of this :+1:

ryanjduffy replied:
Ya, looks like this can be removed. Verified the old bug doesn't repro.

aarontam commented on lib/LabeledTextItem/LabeledTextItem.js:
```diff
@@ -86,21 +72,18 @@ module.exports = kind(
-		this.$.text.setContent(this.text);
-	}
+	components:[
+		// headerContainer required to avoid bad scrollWidth returned in RTL for certain text widths (webkit bug)
```
Wonder if we should re-verify this behavior with Blink while we're at it.

Commit:
removing wrapping DIVs

BHV-1368 added a wrapper around the header to fix marquee issues with
RTL languages. Removing them does not appear to cause a regression on
Blink.

Issue: ENYO-2144
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

Commit:
spot header correctly on key presses in ExpandableInput

Corrects spotting issues when blurring the input either by 5-way or
enter primarily by moving the keyup handler to InputDecorator and
adding dismissOnEnter to Input. This allows the input to be blurred
and therefore Spotlight unfrozen before programmatically spotting the
header.

Also fixed up a couple Spotlight warnings in InputDecorator and
ExpandableListItem.

Issue: ENYO-2144
Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

Commit:
Merge branch '2.6.0-dev' into ENYO-2144-ryanjduffy

Conflicts:
	lib/Accordion.js
	lib/ExpandableInput/ExpandableInput.js
	lib/ExpandableListItem/ExpandableListItem.js
	lib/ExpandablePicker/ExpandablePicker.js
